tests/microgrid/mock_api.py

```diff
@@ -77,10 +78,22 @@ def __init__(
         self._latest_discharge: Optional[PowerLevelParam] = None
 
     def add_component(
-        self, component_id: int, component_category: ComponentCategory.V
+        self,
+        component_id: int,
+        component_category: ComponentCategory.V,
+        inverter_type: InverterType.V = InverterType.TYPE_UNSPECIFIED,
     ) -> None:
         """Add a component to the mock service."""
-        self._components.append(Component(id=component_id, category=component_category))
+        if component_category == ComponentCategory.COMPONENT_CATEGORY_INVERTER:
+            self._components.append(
+                Component(
+                    id=component_id, category=component_category, inverter=inverter_type
+                )
+            )
+        else:
+            self._components.append(
+                Component(id=component_id, category=component_category)
+            )
```

Comment:
If `inverter_type` would be renamed to `component_type` and it would be raw protobuf (as mentioned above), then you will be able to remove all `if` here and use  `_component_type_from_protobuf` (that you implemented in the first commit).